stellarphot/photometry/photometry.py

```diff
@@ -1,138 +1,408 @@
+        A dictionary containing instrumental passband names as keys and
+        AAVSO passband names as values. This is used to rename the passband
+        entries in the output photometry table to be AAVSO standard versus
+        whatever is in the source list.
+
+    fwhm_by_fit : bool, optional (default: True)
+        If ``True``, the FWHM will be calculated by fitting a Gaussian to
+        the star. If ``False``, the FWHM will be calculated by finding the
+        second moments of the light distribution. Default is ``True``.
+
+    fname : str, optional ()
+        Name of the image file on which photometry is being performed.
+
+
     Returns
     -------
-    phot_table : `astropy.table.Table`
-        Astropy table with columns for flux, x/y coordinates of center,
-        RA/dec coordinates of center, sky background per pixel,
-        net flux, aperture and annulus radii used, and flux error.
+
+    phot_table : `stellarphot.PhotometryData`
+        Photometry data for all the locations in which aperture photometry was
+        performed.  This may be a subset of the sources in the sourcelist if
+        locations were too close to the edge of the image or to each other for
+        successful apeture photometry.
+
+    dropped_sources : list
+        This of the star_ids of the sources that fell outside the image or were
+        too close together and did not have photometry performed.
     """
 
-    # check that the outer radius is greater or equal the inner radius
-    # for annulus
+    # Check that the input parameters are valid
+    if not isinstance(ccd_image, CCDData):
+        raise TypeError("ccd_image must be a CCDData object, but it is "
+                        f"{type(ccd_image)}")
+    if not isinstance(sourcelist, SourceListData):
+        raise TypeError("sourcelist must be a SourceListData object, but it is "
+                        f"{type(sourcelist)}")
+    if not isinstance(camera, Camera):
+        raise TypeError(f"camera must be a Camera object, but it is {type(camera)}")
+    if not isinstance(observatory_location, EarthLocation):
+        raise TypeError("observatory_location must be a EarthLocation object, but it "
+                        f"is {type(observatory_location)}")
     if inner_annulus >= outer_annulus:
         raise ValueError("outer_annulus must be greater than inner_annulus")
-
-    # check that the annulus inner radius is greater or equal
-    # the aperture radius
     if aperture_radius >= inner_annulus:
         raise ValueError("inner_radius must be greater than aperture_radius")
+    if (shift_tolerance<=0):
+        raise ValueError("shift_tolerance must be greater than 0 (should be on order "
+                         "of FWHM)")
+    if (max_adu<=0):
+        raise ValueError("max_adu must be greater than 0")
+
+    # Set hot pixels to NaN
+    ccd_image.data[ccd_image.data > max_adu] = np.nan
+
+    # Extract necessary values from sourcelist structure
```

Comment:
I meant why not do this:

```python
star_ids = sourcelist['star_id']  # no .value
xs = sourcelist['xcenter']  # no .value
...
```

Some of those things will have units but it should work out in the end (maybe).

Reply:
Honestly, I was trying to avoid the issue with units in some of the later steps.  For example, when I remove the .value, I get test failures because the selection of `out_of_bounds` fails since I am mixing values with units and no units.